### PR TITLE
[NetFX] DEV | Drop support for Net46 and add support for Net461

### PR DESCRIPTION
**Pull request by JRahnama**
This PR is the first part of the task to drop support for net46 and add support for net461. 
This PR is merely making SDK's features available for other developments in progress for now.

Another PR will be made to address probable upgrades/enhancements in the driver accordingly if there is any.

[Net461 Release notes](https://github.com/Microsoft/dotnet/blob/master/releases/net461/dotnet461-changes.md)

### CODE REVIEW

**Comment by Wraith2** [comment]
Just out of interest what scenarios does this unblock?

**Comment by JRahnama** [comment]
> Just out of interest what scenarios does this unblock?

Although this is a minor version upgrade, there are several improvements in Net461 such bug fixes and a better performance in some areas. This may have some improvements to our test labs as well regarding the netfx failures. Plus it will prevent a huge jump in future when a major version change comes up. I am personally waiting for `4.7.1` which has so many features available such [RuntimeInformation](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation?view=net-5.0)

**Comment by cheenamalhotra** [comment]
> Just out of interest what scenarios does this unblock?

Using Azure.Identity as a dependency for AD authentication :)

File: BUILDGUIDE.md
Comment (karinazhou) [inline BUILDGUIDE.md:203]: Any reason we bump to 462 for the tests?

File: release-notes/2.1/2.1.0-preview1.md
Comment (karinazhou) [inline release-notes/2.1/2.1.0-preview1.md:42]: I don't think we need to touch the notes for these previously released versions since they still support the net46.

File: BUILDGUIDE.md
Comment (JRahnama) [reply BUILDGUIDE.md:203]: This is the sample of `TargetNetFxVersion` since our default is 461 right now, the sample shows a version other than the default. previously, when we had 46 as default and it was 461 showing a different version as a different version.

File: release-notes/2.1/2.1.0-preview1.md
Comment (JRahnama) [reply release-notes/2.1/2.1.0-preview1.md]: done.

File: BUILDGUIDE.md
Comment (karinazhou) [reply BUILDGUIDE.md:203]: I see. Thanks.

File: release-notes/2.1/2.1.0-preview2.md
Comment (karinazhou) [inline release-notes/2.1/2.1.0-preview2.md:123]: Please revert the change here as well

Comment (JRahnama) [reply release-notes/2.1/2.1.0-preview2.md]: done

**Review by karinazhou — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/netcore/src/Common/perf/PerfRunner/PerfRunner.csproj
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/netcore/src/Common/perf/PerfRunner/PerfRunner.csproj:16]: I don't think there's any TargetGroup in this project, it's never used too.
Could you change this to:
```suggestion
  <ItemGroup Condition="'$(TargetFramework)' != 'net461'">
```

File: tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec
Comment (cheenamalhotra) [inline tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec:27]: Let's not touch AKV provider project right now.
This PR shall only target changes in NetFx driver.

File: src/Microsoft.Data.SqlClient/add-ons/Directory.Build.props
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/add-ons/Directory.Build.props:33]: No changes in Add-ons for this PR

File: src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.csproj
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.csproj:9]: Same here. Please revert AKV provider changes.

**Review by cheenamalhotra — COMMENTED** [review]
Minor changes needed:
- Readme.md: Update "Supportability" section.
- BuildGuide.md: Line 72 (Default .NET Framework version is 4.6.)

File: src/Microsoft.Data.SqlClient/netcore/src/Common/perf/PerfRunner/PerfRunner.csproj
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/netcore/src/Common/perf/PerfRunner/PerfRunner.csproj]: done.

File: tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec
Comment (JRahnama) [reply tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec]: done.

File: src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.csproj
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/add-ons/AzureKeyVaultProvider/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.csproj]: done.

File: tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec
Comment (cheenamalhotra) [inline tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec:50]: This entire file needs to be reverted.

Comment (JRahnama) [reply tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec]: Done. I also reverted the package naming.
from AzureKeyVaultProvider\net461\  to AzureKeyVaultProvider\net46\

**Comment by cheenamalhotra** [comment]
@JRahnama 
Please update pipelines to build for net461 and rerun them here - as we need that change eventually.

**Review by cheenamalhotra — APPROVED** [review]
After all Pipelines pass, this can be merged.

**Pull request merged**
